test/stat.c

```diff
@@ -0,0 +1,33 @@
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <errno.h>
+
+int main() {
+  struct stat buf;
+  int rc;
+  while (1) {
+    char procFd[100];
+    snprintf(procFd, sizeof(procFd), "/proc/%d/fd", getpid());
+    procFd[sizeof(procFd)-1] = '\0';
+
+    rc = stat("/dev/tty", &buf);
+    if (rc == -1) return 1;
+    rc = stat("/etc/passwd", &buf);
+    if (rc == -1) return 2;
+    rc = stat(procFd, &buf);
+    if (rc == -1) return 3;
+    rc = stat("/etc/there_is_no_such_file_and_so_rc_is_error", &buf);
+    if (rc != -1 || errno != ENOENT) {
+      printf("Expected 'rc == -1 && errno == ENOENT'; got rc: %d, errno: %d\n",
+             rc, errno);
+      return 4;
+    }
+    char * badAddress = (char *)0xffffffffff000000;
```

Comment:
I made a mistake here.  The definition of `badAddress` should be:

    #if defined(__x86_64__) || defined(__aarch64__)
    const char * badAddress = (char *)0xffffffffff000000;
    #else
    const char * badAddress = (char *)0xff000000;
    #endif```
I'll fix this in the next iteration.